src/branch.c

```diff
@@ -480,18 +505,28 @@ int git_branch_upstream_remote(git_buf *buf, git_repository *repo, const char *r
 		return error;
 
 	if ((error = git_buf_sanitize(buf)) < 0 ||
-	    (error = retrieve_upstream_configuration(buf, cfg, refname, "branch.%s.remote")) < 0)
+	    (error = retrieve_upstream_configuration(buf, cfg, refname, format)) < 0)
 		return error;
 
 	if (git_buf_len(buf) == 0) {
-		git_error_set(GIT_ERROR_REFERENCE, "branch '%s' does not have an upstream remote", refname);
+		git_error_set(GIT_ERROR_REFERENCE, "branch '%s' does not have an upstream %s", refname, format_name);
 		error = GIT_ENOTFOUND;
 		git_buf_clear(buf);
 	}
 
 	return error;
 }
 
+int git_branch_upstream_remote(git_buf *buf, git_repository *repo, const char *refname)
+{
+	return git_branch_upstream_with_format(buf, repo, refname, GIT_BRANCH_UPSTREAM_FORMAT_REMOTE);
+}
+
+int git_branch_upstream_merge(git_buf *buf, git_repository *repo, const char *refname)
+{
+	return git_branch_upstream_with_format(buf, repo, refname, GIT_BRANCH_UPSTREAM_FORMAT_MERGE);
```

Comment:
Instead of adding indirection with the enum and the `switch` and added functions, I'd just pass it in.  Much simpler to just have a function like:

```
static int git_branch_upstream_with_format(git_buf *buf, git_repository *repo, const char *refname, const char *config, const char *name)
```

and then have:

```
int git_branch_upstream_merge(git_buf *buf, git_repository *repo, const char *refname)
{
	return git_branch_upstream_with_format(buf, repo, refname, "branch.%s.merge", "merge");
}
```

No need to worry about match arms in the switch that aren't exercised, etc.

Reply:
Ok, changed.